coa/pkg/apis/v1alpha2/providers/pubsub/memory/memory.go

```diff
@@ -8,30 +8,68 @@ package memory
 
 import (
 	"encoding/json"
+	"strconv"
+	"time"
 
 	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2"
 	contexts "github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/contexts"
 	providers "github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/providers"
+	"github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/utils"
 	"github.com/eclipse-symphony/symphony/coa/pkg/logger"
 )
 
 var log = logger.NewLogger("coa.runtime")
 
+const (
+	DefaultRetryCount      = 5
+	DefaultRetryWaitSecond = 10
+)
+
 type InMemoryPubSubProvider struct {
 	Config      InMemoryPubSubConfig               `json:"config"`
 	Subscribers map[string][]v1alpha2.EventHandler `json:"subscribers"`
 	Context     *contexts.ManagerContext
 }
 
 type InMemoryPubSubConfig struct {
-	Name string `json:"name"`
+	Name                      string `json:"name"`
+	SubscriberRetryCount      int    `json:"subscriberRetryCount"`
+	SubscriberRetryWaitSecond int    `json:"subscriberRetryWaitSecond"`
 }
 
 func InMemoryPubSubConfigFromMap(properties map[string]string) (InMemoryPubSubConfig, error) {
 	ret := InMemoryPubSubConfig{}
 	if v, ok := properties["name"]; ok {
 		ret.Name = v
 	}
+	ret.SubscriberRetryCount = 0
+	if v, ok := properties["subscriberRetryCount"]; ok {
+		val := v
+		if val != "" {
+			n, err := strconv.Atoi(val)
+			if err != nil {
+				return ret, v1alpha2.NewCOAError(err, "invalid int value in the 'SubscriberRetryCount' setting of Memory pub-sub provider", v1alpha2.BadConfig)
+			}
+			ret.SubscriberRetryCount = n
+		}
+	}
+	if ret.SubscriberRetryCount == 0 {
```

Comment:
I would prefer we respect users' valid input retryCount = 0. This means disable any retry. It should be useful when we want to catch one time error especially during development.

Reply:
Fixed. Will throw exception when retryCount is negative.